src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/ApiDescriptionExtensions.cs

```diff
@@ -47,5 +47,26 @@ public static IEnumerable<object> CustomAttributes(this ApiDescription apiDescri
 
             customAttributes = Enumerable.Empty<object>();
         }
+
+        internal static string RelativePathSansParameterConstraints(this ApiDescription apiDescription)
+        {
```

Comment:
Could this be done with a Regex replace? I _feel_ that would be a little cleaner but not sure about perf implications.

Reply:
Perhaps! Admittedly, I'm by now means a regex expert but things get pretty gnarly when you try to capture all of the requirements for the matcher in regex form (matching '?', '=', or ':', handling regex constraints, etc).

In generally, we've avoid parsing routes with regex over in ASP.NET Core. We'll usually roll out token enumerator implementation ([ref](https://github.com/dotnet/aspnetcore/blob/b1560573192f70bfbfa7bb39be51aaafc0c91190/src/Framework/AspNetCoreAnalyzers/src/Analyzers/DelegateEndpoints/DetectMismatchedParameterOptionality.cs#L66)) to handle this.

I do agree that some code comments might make this a little bit easier to read/maintain.